### PR TITLE
Remove GDPR TCF1

### PR DESCRIPTION
**Pull request by bsardo**

### CODE REVIEW

File: gdpr/impl.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ import (
 	"fmt"
 
 	"github.com/prebid/go-gdpr/api"
-	tcf1constants "github.com/prebid/go-gdpr/consentconstants"
-	consentconstants "github.com/prebid/go-gdpr/consentconstants/tcf2"
+	"github.com/prebid/go-gdpr/consentconstants"
+	tcf2ConsentConstants "github.com/prebid/go-gdpr/consentconstants/tcf2"
```
Comment (hhhjort) [inline gdpr/impl.go:9]: Is there a reason for changing this name to tcf2 specific?

Comment (bsardo) [reply gdpr/impl.go:9]: I was trying to maintain the differentiation between the `consentconstants` and `consentconstants/tcf2` packages while removing any reference to tcf1 so I renamed both of these.

**Comment by SyntaxNode** [comment]
I think you may have missed TCF1 references in the metrics. Please look at and update `TCFVersions`, `TCFVersionToValue`, and related tests.

**Comment by SyntaxNode** [comment]
I also just now realized config.PurposeOneTreatement has a spelling mistake.

**Comment by SyntaxNode** [comment]
I think it's also a good idea to add a end to end unit test to make sure that a TCF1 string is not recognized/accepted by the system in a graceful manner.

File: gdpr/impl_test.go [hunk 2]
```diff
@@ -543,7 +557,6 @@ func TestProhibitedPurposeSyncTCF2(t *testing.T) {
 			openrtb_ext.BidderRubicon:  8,
```
[the remark this answers is not in the document]
Comment (bsardo) [reply gdpr/impl_test.go:557]: Ah yep I did that in vendorlist-fetching_test.go but forgot to do it here.

**Comment by bsardo** [comment]
> I also just now realized config.PurposeOneTreatement has a spelling mistake.

I changed this and the name of the config file field that maps to this (`purpose_one_treatement`)which was also misspelled so that makes this a breaking change. Are we ok with that?

**Review by hhhjort — APPROVED** [review]
LGTM

**Review by hhhjort — APPROVED** [review]

**Comment by bsardo** [comment]
I ran the following manual tests to verify TCF1 is handled appropriately and TCF2 still works:

**TCF2 Request:**
```
{
  "id": "some-request-id",
  "test": 1,
  "site": {
    "publisher": { "id": "test"},
    "page": "prebid.org"
  },
  "imp": [{
    "id": "some-impression-id",
    "banner": {
      "format": [{
        "w": 600,
        "h": 500
      }, {
        "w": 300,
        "h": 600
      }]
    },
    "ext": {
      "appnexus": {
        "placementId": 12883451
      }
    }
  }],
  "regs": {
      "ext": {
          "gdpr": 1
      }
  },
  "tmax": 500,
  "user": {
      "id": "my-user-id",
      "buyeruid": "my-buyer-uid",
      "ext": {
          "consent": "CPGWkCaPGWkCaApAAAENAwCAAP_AAAAAAAAAAQAAAAAEAAAAAA"
      }
  }
}
```

**TCF2 Response:**
```
{
    "id": "some-request-id",
    "seatbid": [
        {
            "bid": [
                {
                    "id": "6466123258780423006",
                    "impid": "some-impression-id",
                    "price": 0.01,
                    "adm": "<script src=\"https://nym1-ib.adnxs.com/ab?an_audit=0&test=1&referrer=prebid.org&e=wqT_3QKRDOARBgAAAwDWAAUBCOv0_oUGELDPhLzH1ePJNxj5hJf0mNuYqFoqNgl7FK5H4XqEPxF7FK5H4XqEPxkREgAhEQkAKREJADERCagw-6uSBjjMG0DMG0gCUNCEvzNYsataYABoo79NeKOpBYABAYoBA1VTRJIBAQbwQJgB2ASgAfQDqAEBsAEAuAECwAEDyAEC0AEJ2AEA4AEB8AEAigI7dWYoJ2EnLCA4MDI1OTQsIDE2MjMxNzc4MzUpBRw0cicsIDEwNzk4NzUzNiwyHwD0BQGSAvEDIUtWV3d2QWphN1lNWEVOQ0V2ek1ZQUNDeHExb3dBRGdBUUFSSXpCdFEtNnVTQmxnQVlQX19fXzhQYUFCd0FYZ0JnQUVCaUFFQmtBRUJtQUVCb0FFQnFBRURzQUVBdVFHUjd3cnc0WHFFUDhFQmtlOEs4T0Y2aERfSkFjOVVSZ0FLaE80XzJRRUFBQUFBQUFEd1AtQUJBUFVCQUFBQUFKZ0NBS0FDQUxVQ0FBQUFBTDBDQUFBQUFNQUNBY2dDQWRBQ0FkZ0NBZUFDQU9nQ0FQZ0NBSUFEQVpnREFib0RDVTVaVFRJNk5EUTJNT0FEMUN1SUJBQ1FCQUNZQkFIQkJBQUFBDXoIeVFRDQokQUFBTmdFQVBFRQELCQEgQ0lCZXdpcVFVCQ8YQUR3UDdFRg0NFEFBQURCQh0_AHkVKAFMMigAAFouKADANEFYb0JfQUZ0WWI3QXZnRm92NHdnZ1lEVlZORWlBWUFrQVlCbUFZQW9RWjdGSzVINCFIJDZnR0FiSUdKQWsBXgkBAEIdqwRCawkUAQEAQx0YRExnR0hnLi6aAokBIU1oS083UTb1AShzYXRhSUFRb0FERhltTHpvSlRsbE5Nam8wTkRZd1FOUXJTEekMUEFfVREMDEFBQVcdDABZHQwAYR0MAGMdDJhlQUEu2AIA4ALoqCvqAgpwcmViaWQub3Jn8gIQCgZBRFZfSUQSBjhF1xzyAhEKBkNQRwETOAc2MjEwMzU38gIRCgVDUAETNAg0ODI5NzY5MPICDQoIATsURlJFURIBCRAcUkVNX1VTRVIFEAAnCSCgQ09ERRIbTWFuYWdlZF9CdXllcl81NmE3Y2MwMWUxNGQy8gIMCghDUEcJKhwA8gILCgdDUBUOEAkKBUlPAXsJGgRJTxUaOBMKD0NVU1RPTV9NT0RFTA0kCBoKFjIWACBMRUFGX05BTUUBQQgeCho2HQAIQVNUAT4QSUZJRUQBIRwNCghTUExJVAFN8OUBMIADAIgDAZADAJgDFKADAaoDAMADrALIAwDYAwDgAwDoAwD4AwOABACSBAkvb3BlbnJ0YjKYBACoBACyBAwIABAAGAAgADAAOAC4BADABADIBADSBA4zNTMyI05ZTTI6NDQ2MNoEAggB4AQA8ATQhL8ziAUBmAUAoAX___________8BqgUPc29tZS1yZXF1ZXN0LWlkwAUAyQUAAAAAAADwP9IFCQkAAAAAAAAAANgFAeAFAfAFAfoFBAgAEACQBgCYBgCyBjJDUEdXa0NhUEdXa0NhQXBBQUFFTkF3Q0FBUF9BQUFRAQFJME3GELgGAcEGBVUsAPA_0AbbDdoGFgoQCREZAQHgTOAGAfIGAggAgAcBiAcAoAcBugcOLvoAKEAAyAejqQXSBw0JETkBNwjaBwYJJijgBwDqBwIIAPAHAA..&s=4e690c7b908860607c98f50aa62b65c0ba1d4dba&pp=${AUCTION_PRICE}\"></script>",
                    "adid": "107987536",
                    "adomain": [
                        "appnexus.com"
                    ],
                    "iurl": "https://nym1-ib.adnxs.com/cr?id=107987536",
                    "cid": "3532",
                    "crid": "107987536",
                    "w": 600,
                    "h": 500,
                    "ext": {
                        "appnexus": {
                            "auction_id": 4004701365052975000,
                            "bid_ad_type": 0,
                            "bidder_id": 2,
                            "brand_id": 1
                        },
                        "prebid": {
                            "type": "banner",
                            "video": {
                                "duration": 0,
                                "primary_category": ""
                            }
                        }
                    }
                }
            ],
            "seat": "appnexus"
        }
    ],
    "cur": "USD",
    "ext": {
        "debug": {
            "httpcalls": {
                "appnexus": [
                    {
                        "uri": "http://ib.adnxs.com/openrtb2",
                        "requestbody": "{\"id\":\"some-request-id\",\"imp\":[{\"id\":\"some-impression-id\",\"banner\":{\"format\":[{\"w\":600,\"h\":500},{\"w\":300,\"h\":600}],\"w\":600,\"h\":500},\"ext\":{\"appnexus\":{\"placement_id\":12883451}}}],\"site\":{\"page\":\"prebid.org\",\"publisher\":{\"id\":\"test\"},\"ext\":{\"amp\":0}},\"device\":{\"ua\":\"PostmanRuntime/7.26.8\"},\"user\":{\"id\":\"my-user-id\",\"buyeruid\":\"my-buyer-uid\",\"ext\":{\"consent\":\"CPGWkCaPGWkCaApAAAENAwCAAP_AAAAAAAAAAQAAAAAEAAAAAA\"}},\"test\":1,\"at\":1,\"tmax\":500,\"source\":{\"tid\":\"d391092c-c02b-48f2-a00a-5a0a2261c659\"},\"regs\":{\"ext\":{\"gdpr\":1}},\"ext\":{\"prebid\":{},\"appnexus\":{\"hb_source\":5}}}",
                        "requestheaders": {
                            "Accept": [
                                "application/json"
                            ],
                            "Content-Type": [
                                "application/json;charset=utf-8"
                            ]
                        },
                        "responsebody": "{\"id\":\"some-request-id\",\"seatbid\":[{\"bid\":[{\"id\":\"6466123258780423006\",\"impid\":\"some-impression-id\",\"price\":0.010000,\"adid\":\"107987536\",\"adm\":\"<script src=\\\"https:\/\/nym1-ib.adnxs.com\/ab?an_audit=0&test=1&referrer=prebid.org&e=wqT_3QKRDOARBgAAAwDWAAUBCOv0_oUGELDPhLzH1ePJNxj5hJf0mNuYqFoqNgl7FK5H4XqEPxF7FK5H4XqEPxkREgAhEQkAKREJADERCagw-6uSBjjMG0DMG0gCUNCEvzNYsataYABoo79NeKOpBYABAYoBA1VTRJIBAQbwQJgB2ASgAfQDqAEBsAEAuAECwAEDyAEC0AEJ2AEA4AEB8AEAigI7dWYoJ2EnLCA4MDI1OTQsIDE2MjMxNzc4MzUpBRw0cicsIDEwNzk4NzUzNiwyHwD0BQGSAvEDIUtWV3d2QWphN1lNWEVOQ0V2ek1ZQUNDeHExb3dBRGdBUUFSSXpCdFEtNnVTQmxnQVlQX19fXzhQYUFCd0FYZ0JnQUVCaUFFQmtBRUJtQUVCb0FFQnFBRURzQUVBdVFHUjd3cnc0WHFFUDhFQmtlOEs4T0Y2aERfSkFjOVVSZ0FLaE80XzJRRUFBQUFBQUFEd1AtQUJBUFVCQUFBQUFKZ0NBS0FDQUxVQ0FBQUFBTDBDQUFBQUFNQUNBY2dDQWRBQ0FkZ0NBZUFDQU9nQ0FQZ0NBSUFEQVpnREFib0RDVTVaVFRJNk5EUTJNT0FEMUN1SUJBQ1FCQUNZQkFIQkJBQUFBDXoIeVFRDQokQUFBTmdFQVBFRQELCQEgQ0lCZXdpcVFVCQ8YQUR3UDdFRg0NFEFBQURCQh0_AHkVKAFMMigAAFouKADANEFYb0JfQUZ0WWI3QXZnRm92NHdnZ1lEVlZORWlBWUFrQVlCbUFZQW9RWjdGSzVINCFIJDZnR0FiSUdKQWsBXgkBAEIdqwRCawkUAQEAQx0YRExnR0hnLi6aAokBIU1oS083UTb1AShzYXRhSUFRb0FERhltTHpvSlRsbE5Nam8wTkRZd1FOUXJTEekMUEFfVREMDEFBQVcdDABZHQwAYR0MAGMdDJhlQUEu2AIA4ALoqCvqAgpwcmViaWQub3Jn8gIQCgZBRFZfSUQSBjhF1xzyAhEKBkNQRwETOAc2MjEwMzU38gIRCgVDUAETNAg0ODI5NzY5MPICDQoIATsURlJFURIBCRAcUkVNX1VTRVIFEAAnCSCgQ09ERRIbTWFuYWdlZF9CdXllcl81NmE3Y2MwMWUxNGQy8gIMCghDUEcJKhwA8gILCgdDUBUOEAkKBUlPAXsJGgRJTxUaOBMKD0NVU1RPTV9NT0RFTA0kCBoKFjIWACBMRUFGX05BTUUBQQgeCho2HQAIQVNUAT4QSUZJRUQBIRwNCghTUExJVAFN8OUBMIADAIgDAZADAJgDFKADAaoDAMADrALIAwDYAwDgAwDoAwD4AwOABACSBAkvb3BlbnJ0YjKYBACoBACyBAwIABAAGAAgADAAOAC4BADABADIBADSBA4zNTMyI05ZTTI6NDQ2MNoEAggB4AQA8ATQhL8ziAUBmAUAoAX___________8BqgUPc29tZS1yZXF1ZXN0LWlkwAUAyQUAAAAAAADwP9IFCQkAAAAAAAAAANgFAeAFAfAFAfoFBAgAEACQBgCYBgCyBjJDUEdXa0NhUEdXa0NhQXBBQUFFTkF3Q0FBUF9BQUFRAQFJME3GELgGAcEGBVUsAPA_0AbbDdoGFgoQCREZAQHgTOAGAfIGAggAgAcBiAcAoAcBugcOLvoAKEAAyAejqQXSBw0JETkBNwjaBwYJJijgBwDqBwIIAPAHAA..&s=4e690c7b908860607c98f50aa62b65c0ba1d4dba&pp=${AUCTION_PRICE}\\\"><\/script>\",\"adomain\":[\"appnexus.com\"],\"iurl\":\"https:\/\/nym1-ib.adnxs.com\/cr?id=107987536\",\"cid\":\"3532\",\"crid\":\"107987536\",\"h\":500,\"w\":600,\"ext\":{\"appnexus\":{\"brand_id\":1,\"auction_id\":4004701365052975024,\"bidder_id\":2,\"bid_ad_type\":0}}}],\"seat\":\"3532\"}],\"bidid\":\"2228983401110947092\",\"cur\":\"USD\"}",
                        "status": 200
                    }
                ]
            },
            "resolvedrequest": {
                "id": "some-request-id",
                "imp": [
                    {
                        "id": "some-impression-id",
                        "banner": {
                            "format": [
                                {
                                    "w": 600,
                                    "h": 500
                                },
                                {
                                    "w": 300,
                                    "h": 600
                                }
                            ]
                        },
                        "ext": {
                            "appnexus": {
                                "placementId": 12883451
                            }
                        }
                    }
                ],
                "site": {
                    "page": "prebid.org",
                    "publisher": {
                        "id": "test"
                    },
                    "ext": {
                        "amp": 0
                    }
                },
                "device": {
                    "ua": "PostmanRuntime/7.26.8"
                },
                "user": {
                    "id": "my-user-id",
                    "buyeruid": "my-buyer-uid",
                    "ext": {
                        "consent": "CPGWkCaPGWkCaApAAAENAwCAAP_AAAAAAAAAAQAAAAAEAAAAAA"
                    }
                },
                "test": 1,
                "at": 1,
                "tmax": 500,
                "source": {
                    "tid": "d391092c-c02b-48f2-a00a-5a0a2261c659"
                },
                "regs": {
                    "ext": {
                        "gdpr": 1
                    }
                }
            }
        },
        "responsetimemillis": {
            "appnexus": 76
        },
        "tmaxrequest": 500,
        "prebid": {
            "auctiontimestamp": 1623177835010
        }
    }
}
```

**Comment by bsardo** [comment]
**TCF 1 Request**
```
{
  "id": "some-request-id",
  "test": 1,
  "site": {
    "publisher": { "id": "test"},
    "page": "prebid.org"
  },
  "imp": [{
    "id": "some-impression-id",
    "banner": {
      "format": [{
        "w": 600,
        "h": 500
      }, {
        "w": 300,
        "h": 600
      }]
    },
    "ext": {
      "appnexus": {
        "placementId": 12883451
      }
    }
  }],
  "regs": {
      "ext": {
          "gdpr": 1
      }
  },
  "tmax": 500,
  "user": {
      "id": "my-user-id",
      "buyeruid": "my-buyer-uid",
      "ext": {
          "consent": "BONV8oqONXwgmADACHENAO7pqzAEBAAAAAgAAAAA"
      }
  }
}
```

**TCF 1 Response**
```
{
    "id": "some-request-id",
    "nbr": 2,
    "ext": {
        "debug": {
            "resolvedrequest": {
                "id": "some-request-id",
                "imp": [
                    {
                        "id": "some-impression-id",
                        "banner": {
                            "format": [
                                {
                                    "w": 600,
                                    "h": 500
                                },
                                {
                                    "w": 300,
                                    "h": 600
                                }
                            ]
                        },
                        "ext": {
                            "appnexus": {
                                "placementId": 12883451
                            }
                        }
                    }
                ],
                "site": {
                    "page": "prebid.org",
                    "publisher": {
                        "id": "test"
                    },
                    "ext": {
                        "amp": 0
                    }
                },
                "device": {
                    "ua": "PostmanRuntime/7.26.8"
                },
                "user": {
                    "id": "my-user-id",
                    "buyeruid": "my-buyer-uid",
                    "ext": {
                        "consent": "BONV8oqONXwgmADACHENAO7pqzAEBAAAAAgAAAAA"
                    }
                },
                "test": 1,
                "at": 1,
                "tmax": 500,
                "source": {
                    "tid": "9576c965-0833-49eb-a9b0-f52176bca820"
                },
                "regs": {
                    "ext": {
                        "gdpr": 1
                    }
                }
            }
        },
        "tmaxrequest": 500,
        "prebid": {
            "auctiontimestamp": 1623177966710
        }
    }
}
```

File: gdpr/impl_test.go [hunk 3]
```diff
@@ -787,3 +798,21 @@ func TestAllowActivitiesTCF2BidRequests(t *testing.T) {
...
+		},
+	}
+
+	bidReq, passGeo, passID, err := perms.AuctionActivitiesAllowed(context.Background(), bidderAllowedByConsent, "", SignalYes, tcf1Consent, false)
+
+	assert.Nil(t, err, "TCF1 consent - no error returned")
+	assert.Equal(t, false, bidReq, "TCF1 consent - bid request not allowed")
+	assert.Equal(t, false, passGeo, "TCF1 consent - passing geo not allowed")
+	assert.Equal(t, false, passID, "TCF1 consent - passing id not allowed")
+}
```
Comment (SyntaxNode) [inline gdpr/impl_test.go:818]: Nice. I like this tests that a TCF1 consent string is treated the same as an invalid TCF2 consent string.

**Review by SyntaxNode — APPROVED** [review]

**Pull request merged**